### PR TITLE
[gulp-lesshint] Add type definition for gulp-lesshint

### PR DESCRIPTION
**Pull request by martin-badin**
Documentation: https://github.com/lesshint/gulp-lesshint
Discussion: https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/58110

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test gulp-lesshint`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/gulp-lesshint/gulp-lesshint-tests.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import gulp = require('gulp');
+import lesshint = require('gulp-lesshint');
+import assert = require('assert');
```
Comment (peterblazejewicz) [inline types/gulp-lesshint/gulp-lesshint-tests.ts:3]:
```suggestion
```

File: types/gulp-lesshint/gulp-lesshint-tests.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+gulp.task('lint', () => {
+    return gulp
+        .src('./src/*.less')
+        .pipe(
+            lesshint({ configPath: './path', maxWarnings: 1 })
+                .on('data', file => {
+                    assert.strictEqual(file.lesshint.success, false);
```
Comment (peterblazejewicz) [inline types/gulp-lesshint/gulp-lesshint-tests.ts:11]:
```suggestion
                    file.lesshint.success; // $ExpectType boolean;
```
(but you would have to use 'any' I think, unless you extend your interface like:

```ts
declare namespace gulpLesshint {
    interface Options {
        configPath?: string;
        maxWarnings?: number;
    }

    interface LessHintFile {
        lesshint: {
            success: boolean;
            resultCount: number;
            results: {
                column: number;
                file: string;
                fullPath: string;
                line: number;
                linter: string;
                message: string;
                severity: string;
            };
        };
        [key: string]: any;
    }

    interface Plugin {
        (options?: Options): {
            on(event: 'data', listener: (file: LessHintFile) => void): Transform;
        } & Transform;

        reporter(reporter?: string): Transform;
        failOnError(): Transform;
        failOnWarning(): Transform;
    }
}
```

File: types/gulp-lesshint/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+declare namespace gulpLesshint {
+    interface Options {
+        configPath?: string;
+        maxWarnings?: number;
+    }
+
+    interface Plugin {
+        (options?: Options): Transform;
+
```
Comment (peterblazejewicz) [inline types/gulp-lesshint/index.d.ts:18]:
```suggestion
```

File: types/gulp-lesshint/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        reporter(reporter?: string): Transform;
+        failOnError(): Transform;
+        failOnWarning(): Transform;
+    }
+}
+
+declare const GulpLesshint: gulpLesshint.Plugin;
+
+export = GulpLesshint;
```
Comment (peterblazejewicz) [inline types/gulp-lesshint/index.d.ts:27]:
```suggestion
declare const lesshintPlugin: gulpLesshint.Plugin;

export = lesshintPlugin;
```

File: types/gulp-lesshint/gulp-lesshint-tests.ts (hunk 2 above)
Comment (martin-badin) [reply types/gulp-lesshint/gulp-lesshint-tests.ts]: Thanks, I'll use it

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@martin-badin thx!

**Comment by martin-badin** [comment]
Ready to merge

**Pull request merged**